src/generate.dart

```diff
@@ -74,90 +85,133 @@ class DiagramGenerator {
 
   /// The temporary directory used to write screenshots and cropped out images
   /// into.
-  Directory tmpDir;
+  Directory temporaryDirectory;
 
-  static String get projectDir =>
-      path.joinAll(path.split(path.fromUri(Platform.script))..removeLast());
+  static String get projectDir {
+    return path.joinAll(path.split(path.fromUri(Platform.script))..removeLast());
+  }
 
   Future<Null> generateDiagram() async {
     await _collectScreenshot();
     await _optimizeImages();
     if (cleanup) {
-      await tmpDir.delete(recursive: true);
+      await temporaryDirectory.delete(recursive: true);
     }
   }
 
-  bool _captureScreenshot(String outputName) {
-    print('Capturing screenshot into $outputName.');
-    ProcessResult processResult = processRunner(flutterCommand, ['screenshot', '--out=$outputName'],
-        workingDirectory: projectDir);
-    if (processResult.exitCode != 0) {
-      print("Failed to run command '$flutterCommand screenshot --out=$outputName' in ${tmpDir.path}:\n${processResult.stderr}");
-      return false;
+  static const Duration timeout = const Duration(seconds: 60);
+
+  Future<List<String>> runFlutter({
+    @required List<String> args,
+    @required String until,
+    AsyncCallback then,
+  }) async {
+    final Process process = await processStarter(flutterCommand, args, workingDirectory: projectDir);
```

Comment:
This could use the ProcessManager instead, from [package:process](https://pub.dartlang.org/packages/process).  This allows us to mock the ProcessManager instead of needing to mock the start and run methods and have the typedefs, etc.

Reply:
Unless you are particularly concerned about this, I suggest we leave it as-is, since this code will all end up going by the wayside when we move to the `flutter drive` approach we discussed earlier today.

Reply:
OK, works for me.